openfpga/src/repack/repack_option.cpp

```diff
@@ -46,6 +46,56 @@ bool RepackOption::is_pin_ignore_global_nets(const std::string& pb_type_name,
   return false;
 }
 
+bool RepackOption::net_is_specified_to_be_ignored(std::string cluster_net_name,
+                                                  std::string pb_type_name,
+                                                  const BasicPort& port) const {
+  auto result = design_constraints_.ignore_net_pin_map().find(cluster_net_name);
+  if (result == design_constraints_.ignore_net_pin_map().end()) {
+    /* Not found, return false */
+    return false;
+  } else {
+    /* Split the content using a tokenizer */
+    auto pin_ctx_to_parse =
+      design_constraints_.ignore_net_pin_map()[cluster_net_name];
+    for (auto pin_ctx_to_parse_iter : pin_ctx_to_parse) {
```

Comment:
I suggest to do the parsing when loading data into repack design constraints. It is clear and also early to report errors. In your current version, such error is seen when repack command is called. This is too late (consider a case where implementation took 1 hour to finish).

Reply:
This can be done as part of dc file legalization. We can add such legalization in dc file generation (which will be handled in generating dc file, in another project).

Reply:
parsed the pin_ctx to filter out invalid format 
![image](https://github.com/lnis-uofu/OpenFPGA/assets/145083116/5d518309-6cee-41fd-916a-4c9bca21190c)